projects/projects_model.js

```diff
@@ -0,0 +1,74 @@
+const db = require('../data/db-config');
+
+module.exports = {
+  find,
+  findById,
+  findResource,
+  findResourceById,
+  insert,
+  insertResource,
+  findTasks,
+  findTaskById,
+  insertTask
+}
+
+function find() {
+  return db('projects')
+}
+
+function findById(id) {
+  return db('projects')
+    .where({id})
+    .first()
+}
+
+function insert(data) {
+  return db('projects').insert(data, 'id')
+    .then(([id]) => {
+      return findById(id)
+    })
+}
+
+function findResource(projectId) {
+  return db('resources')
+    .join('project_resources', 'resources.id', 'project_resources.resource_id')
+    .where({'project_resources.project_id': projectId})
+}
+
+function findResourceById(id) {
+  return db('resources')
+    .where(({id}))
+}
+
+function insertResource(projectId, resource) {
+  return db('resources').insert(resource)
+    .then(([id]) => {
+      return db('project_resources').insert({
+        project_id: projectId,
+        resource_id: id,
+      })
+    })
+    .catch(() => {
+      return findResourceById(id)
+    })
+}
+
+function findTasks(projectId) {
+  return db('tasks')
+    .join('projects', 'tasks.project_id', 'projects.id')
+    .where({'tasks.project_id': projectId})
+    .select('tasks.*', 'projects.name as project_name', 'projects.description as project_description')
+}
+
+function findTaskById(id) {
+  return db('tasks')
+    .where({id})
+    .first()
+}
+
+function insertTask(projectId, task) {
+  return db('tasks').insert({project_id: projectId, ...task})
+    .then(([id]) => {
+      return findTaskById(id)
+    })
```

Comment:
Small note: Make sure that you spread first, and then add extra properties afterwards.

If we had objects like this:
```
const task = {
    description: "feed the fish",
    project_id: 27 // We don't want this value
}
const projectId = 35 // We want this value

const newObj = {project_id: projectId, ...task}

console.log(newObj) // -> { description: "feed the fish", project_id: 27 }
```
Because we spread `tasks` after `project_id`, we ended up with a project id of 27, instead of 35.

What we really want to do here is:
```
const task = {
    description: "feed the fish",
    project_id: 27
}
const projectId = 35

const newObj = { ...task, project_id: projectId }

console.log(newObj) // -> { description: "feed the fish", project_id: 35 }
```

This way, the project id is the correct id we're looking for 👍

Reply:
Thanks, for that and I will definitely remember for next time!!